### ISSUE TITLE
Not getting detailed sentry alert message

### ISSUE THREAD

**Issue by thisis4github** [Its-Alex/sentry-mattermost-sidecar#4]
Hi,

Not getting detailed sentry alert in the Mattermost channel. Project and environment details are missing in the message. Is there any additional setup needed to obtain the complete information in the alert message?

![image](https://github.com/Its-Alex/sentry-mattermost-sidecar/assets/141025371/8b835176-ba80-4f72-a360-44539ff125f2)

FYI: we are using Sentry 8.15.0 and latest Mattermost version.

**Comment by bfontaine** [comment]
I’m using this tool without issue; I get all the details in the notification. It worked with the legacy "webhooks" integration, and it still works with the new custom integration setup.
![webhook setup](https://github.com/Its-Alex/sentry-mattermost-sidecar/assets/1334295/09b64c15-f861-406c-9cf0-1c205af0565c)
![custom integration setup](https://github.com/Its-Alex/sentry-mattermost-sidecar/assets/1334295/39a97f9e-ab9d-44b3-953a-9b568769c1f0)

**Comment by Its-Alex** [comment]
@thisis4github Hello :wave: Can you give me some useful informations that will help me try on my hands:

1. Sentry version that you use
1. Mattermost version that you use
1. Sentry-mattermost-sidecar version that you use
1. Can you give me an example of the error that you throw? A way to throw it myself?

**Comment by bfontaine** [comment]
After a while of not using this integration I set it up again and I get empty notifications as well:
![](https://github.com/user-attachments/assets/93f69bb0-ec03-40b3-9bf5-22ad6bdf04ba)

* Sentry: Cloud version
* Mattermost 9.11.1 (latest extended-support release)
* Sentry-mattermost-sidecar 1.0.4 (latest version)

**Comment by AKhMongoloid** [comment]
Same problem.

- Sentry 24.2.0 (Self-Hosted)
- Mattermost: 9.4.2
- Sentry-mattermost-sidecar 1.0.4 (latest version)

**Comment by Its-Alex** [comment]
I will try to do something this week, thanks for these reports :pray:

**Comment by Its-Alex** [comment]
@bfontaine @AKhMongoloid I can't reproduce the error using Mattermost `9.4.2` and `9.11.1` and Sentry selfhosted `24.2.0`:

![image](https://github.com/user-attachments/assets/7fc3f90a-2514-481c-968e-bd70e6a27bee)

Can you please provide me some additionals informations:

- Can you give me the body (at least the format) of the request that is send by Sentry to Sentry-mattermost-sidecar? It should match something like [this one](https://github.com/Its-Alex/sentry-mattermost-sidecar/blob/main/scripts/test-request.sh#L7-L16)
- Can you try to trigger a manual exceptions in your mattermost channel using curl? You can find the format in the [mattermost documentation](https://developers.mattermost.com/integrate/reference/message-attachments/)
- Can you give me the URL used on sentry to fetch sentry-mattermost-sidecar?

**Comment by Its-Alex** [comment]
@thisis4github If you have some useful information, can you give it to me too?

**Comment by bfontaine** [comment]
* I use the Cloud Sentry. Here is the format of the test notification: https://privatebin.net/?66a55b5011b4fa43#H2XLeYsY85X9W8gkMN7Wn3wB9BvNU9gNvgaZ79zEBe2H
* I’m not sure what you mean; I already use the Mattermost API to post message attachments without issue
* https://mydomain.example.com/sentry-alerts

**Comment by Its-Alex** [comment]
@bfontaine Okay this sidecar is made for custom integration, I will try to see if it can support both. Thanks for you're informations!

**Comment by bfontaine** [comment]
Aaah ok I didn’t know that there was a difference. Indeed, if I use the legacy integration it works!

**Comment by AKhMongoloid** [comment]
@Its-Alex  Sorry for my very late answer. Fixed it as you pointed out, and everything is working now. My mistake: I was creating a new integration, but I should have added it to the webhooks. Thank you for your effort.

**Comment by bfontaine** [comment]
@Its-Alex I use only issues, but with the legacy integration it works as expected.

**Comment by Its-Alex** [comment]
@thisis4github @bfontaine I've updated the code to support custom integration from sentry (for issues and errors) but for now I can't get the link of the issue from the custom integrations.

I will try to add the link and publish a new release

**Comment by robbertu** [comment]
> [@bfontaine](https://github.com/bfontaine) What is the most used event for you today between all [webhooks events](https://docs.sentry.io/organization/integrations/integration-platform/webhooks/)? I will implement then and try to do in order of what is the most used

For us this would be [Issue Alerts](https://docs.sentry.io/organization/integrations/integration-platform/webhooks/issue-alerts/). It would be great to be able to send a notification to Mattermost when an alert is triggered.

**Comment by Its-Alex** [comment]
@robbertu Tell me if I'm wrong, but that is what is used in [custom integration of Sentry](https://docs.sentry.io/organization/integrations/integration-platform/webhooks/#:~:text=sentry's%20integration%20platform%20webhooks), isn't it?

I mention it [in the README](https://github.com/Its-Alex/sentry-mattermost-sidecar/blob/main/README.md?plain=1#L9). If you try the `latest` tag, it should work as expected, but without an issues link, since I can't find it in Sentry's payload.

**Comment by robbertu** [comment]
@Its-Alex True, but you also mentioned the Issue Alerts as Custom Integration in the roadmap, instead of the legacy integration. I was hoping for that. But I will also take a look at the legacy integration, thanks.

**Comment by Its-Alex** [comment]
@robbertu It doesn't seem clear, but previously the project worked with legacy integrations. Since these integrations weren't considered "legacy" when this service was created, the readme didn't mention them.

In a [recent PR](https://github.com/Its-Alex/sentry-mattermost-sidecar/pull/5), I added support for issue alerts from custom integrations. Now the [readme shows support for both legacy integrations and Issue Alerts as Custom Integration](https://github.com/Its-Alex/sentry-mattermost-sidecar?tab=readme-ov-file#roadmap).

If you want to test the functionality, you can use the `latest` tag since no release has been made yet.

**Comment by robbertu** [comment]
@Its-Alex Maybe I wasn't specific enough. I was talking about the Issue Alerts "triggered" event. So when an Issue Alert configured in Sentry is triggered, I need a notification about this event in Mattermost. I think this is currently only possible with the legacy integration, right?

**Comment by Its-Alex** [comment]
@robbertu @bfontaine  Hello 👋

The all alerts concerning issues (legacy and current) are now available in `1.1.0`. You can try it and open a PR if there is a problem.

I will close this issue since all new webhook integration for issues and alert in sentry is now working

**Comment by bfontaine** [comment]
Thank you @Its-Alex!
